crossengage/client.py

```diff
@@ -99,7 +141,19 @@ def delete_user(self, user):
         :return: json dict response, for example: {"status_code": 200}
         """
         self.request_url = self.API_URL + self.USER_ENDPOINT + user['id']
-        return self.__create_request(payload=user, request_type=self.REQUEST_DELETE)
+        return self.__create_request(payload=user, request_type=self.REQUEST_DELETE, headers=self.default_headers)
+
+    def delete_user_async(self, user):
+        # type: (dict) -> dict
+        """
+        Delete User given its id.
+        :param user: dict of payload (id)
+        :return: json dict response, for example:
+            {"status_code": 202, "trackingId": "2e312089-a987-45c6-adbd-b904bc4dfc97"}
+        """
+        headers = update_dict(self.default_headers, {self.API_VERSION_HEADER: self.API_VERSIONS["v2"]})
```

Comment:
It's possible but how do we know if it should be called since we always pass the headers as param? Also, it's the outer method (e.g `delete_user_async`) who knows about the api version.

Reply:
I mean just avoid code duplication:

```python
self.__create_request(payload=user, request_type=self.REQUEST_DELETE, headers={self.API_VERSION_HEADER: self.API_VERSIONS["v2"]})
```

```python
def __create_request(self, payload, request_type, headers):
... .... ... 
    r = self.requests.put(self.request_url, data=json.dumps(payload), headers=update_dict(self.default_headers , headers))
```